github-pullrequest-plugin/src/test/java/org/jenkinsci/plugins/github/pullrequest/trigger/check/PullRequestToCauseConverterTest.java

```diff
@@ -116,4 +126,32 @@ public void shouldReturnCauseOnWhen1OpenedWeGetSecondOneAndHaveEvents() throws E
         assertThat("reason in cause", cause.getReason(), containsString("open"));
         assertThat("pr num in cause", cause.getNumber(), is(2));
     }
+
+    /**
+     * Test trigger configuration of:
+     *
+     *     1.) Skip PR if label is not present (when label is not present)
+     *     2.) Cause PR if commit changed (when commit has changed)
+     *
+     * Expected result is that the PR should be skipped. No causes should be
+     * identified.
+     */
+    @Test
+    public void shouldSkipEvenThoughNewCommit() throws Exception {
+        GHCommitPointer commitPtr = mock(GHCommitPointer.class);
+        when(local.getPulls()).thenReturn(ImmutableMap.of(3, localPR));
+        when(localPR.getHeadSha()).thenReturn("this is not the sha you are looking for");
+        when(remotePR.getNumber()).thenReturn(3);
+        when(remotePR.getState()).thenReturn(GHIssueState.OPEN);
+        when(remotePR.getHead()).thenReturn(commitPtr);
+        when(trigger.getEvents()).thenReturn(asList(
+                new GitHubPRLabelNotExistsEvent(new GitHubPRLabel("notfound"), true),
+                new GitHubPRCommitEvent()));
+
+        List<GitHubPRCause> causes = from(ImmutableSet.of(remotePR))
+                .transform(toGitHubPRCause(local, tlRule.getListener(), trigger))
+                .filter(notNull()).toList();
+
+        assertThat("no cause", causes, hasSize(0));
```

Comment:
That's because monolite logic was replaced with streams. See https://github.com/jenkinsci/github-integration-plugin/pull/22 

As i understand `.apply` `.filter` etc should filter causes...

Let me remember.
First place with logic works in terms of single PR should stop evaluating PRs on first non-null cause for this pr. Cause would be either skip at all pr, either trigger it.

Second: list of casuses from all PRs is filtered and only non skip causes triggering the build.